python_modules/libraries/dagster-dbt/dagster_dbt/core/resources_v2.py

```diff
@@ -633,6 +646,98 @@ def is_successful(self) -> bool:
 
         return self.process.wait() == 0
 
+    def _stream_asset_events(
+        self,
+    ) -> Iterator[DbtDagsterEventType]:
+        """Stream the dbt CLI events and convert them to Dagster events."""
+        for event in self.stream_raw_events():
+            yield from event.to_default_asset_events(
+                manifest=self.manifest,
+                dagster_dbt_translator=self.dagster_dbt_translator,
+                context=self.context,
+                target_path=self.target_path,
+            )
+
+    def _get_dbt_resource_props_from_event(self, event: DbtDagsterEventType) -> Dict[str, Any]:
+        unique_id = cast(TextMetadataValue, event.metadata["unique_id"]).text
+        return check.not_none(self.manifest["nodes"].get(unique_id))
+
+    def _attach_post_materialization_metadata(
+        self,
+        event: DbtDagsterEventType,
+    ) -> DbtDagsterEventType:
+        """Threaded task which runs any postprocessing steps on the given event before it's
+        emitted to user code.
+
+        This is used to, for example, query the row count of a table after it has been
+        materialized by dbt.
+        """
+        adapter = check.not_none(self.adapter)
+
+        dbt_resource_props = self._get_dbt_resource_props_from_event(event)
+        is_view = dbt_resource_props["config"]["materialized"] == "view"
+
+        # Avoid counting rows for views, since they may include complex SQL queries
+        # that are costly to execute. We can revisit this in the future if there is
+        # a demand for it.
+        if is_view:
+            return event
+
+        # If the adapter is DuckDB, we need to wait for the dbt CLI process to complete
+        # so that the DuckDB lock is released
```

Comment:
Let's add some more comments about why this is an edge case for only DuckDB.

Other adapters should be fine in supporting multiple connections.